library/std/src/io/entropy.rs

```diff
@@ -0,0 +1,125 @@
+use super::{BorrowedBuf, BorrowedCursor, Read, Result};
+use crate::sys::entropy as sys;
+
+/// A reader which returns random bytes from the system entropy source.
+///
+/// This struct is generally created by calling [`entropy()`]. Please
+/// see the documentation of [`entropy()`] for more details.
+#[derive(Debug)]
+#[unstable(feature = "io_entropy", issue = "none")]
+pub struct Entropy {
+    insecure: bool,
```

Comment:
This could probably use `sys::Entropy` directly here (and define a `set_insecure` method for it)